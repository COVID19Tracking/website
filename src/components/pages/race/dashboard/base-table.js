import React from 'react'
import { StateTable, StateTableHeader, StateTableBody } from './table'

const BaseTable = ({
  data,
  type,
  noPositives,
  noDeaths,
  noTests,
  noHospitalizations,
  rows,
}) => {
  return (
    <StateTable>
      <StateTableHeader
        groupTitle={type}
        noPositives={noPositives}
        noDeaths={noDeaths}
        noTests={noTests}
        noHospitalizations={noHospitalizations}
      />
      <StateTableBody
        noPositives={noPositives}
        noDeaths={noDeaths}
        noTests={noTests}
        noHospitalizations={noHospitalizations}
        state={data.name}
        stateAbbr={data.state}
        type={type}
        rows={rows}
      />
    </StateTable>
  )
}

export default BaseTable
