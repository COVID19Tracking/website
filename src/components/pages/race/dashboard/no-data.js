import React from 'react'
import noDataStyles from './no-data.module.scss'
import alertBang from '~images/race-dashboard/alert-bang-white.svg'

const NoData = ({ stateName }) => (
  <div className={noDataStyles.container}>
    <table className={noDataStyles.table} aria-hidden role="presentation">
      <tbody>
        <tr>
          <td />
          <td />
          <td />
          <td />
        </tr>
        <tr>
          <td />
          <td />
          <td />
          <td />
        </tr>
        <tr>
          <td />
          <td />
          <td />
          <td />
        </tr>
        <tr>
          <td />
          <td />
          <td />
          <td />
        </tr>
        <tr>
          <td />
          <td />
          <td />
          <td />
        </tr>
        <tr>
          <td />
          <td />
          <td />
          <td />
        </tr>
      </tbody>
    </table>
    <div className={noDataStyles.alert}>
      <div className={noDataStyles.title}>
        <img src={alertBang} alt="Alert icon" />
        <span>No Data</span>
      </div>
      <div className={noDataStyles.content}>
        <p>{stateName} does not share data about race or ethnicity. </p>
      </div>
    </div>
  </div>
)

export default NoData
