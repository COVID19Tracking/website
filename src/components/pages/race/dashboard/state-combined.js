import React from 'react'
import HeaderSorter from './header-sorter'
import TableNotes from './table-notes'
import PercentageOverview from './percentage-overview'
import anhpiNotes from './anhpi-notes'
import cautionNotes from './caution-notes'
import RaceTable from './race-table'
import stateStyle from './state.module.scss'

const StateCombined = ({ state }) => {
  const stateData = state
  let notes = {
    otherDeath: stateData.otherDeathNotes,
    otherPos: stateData.otherPosNotes,
    whiteDeath: stateData.whiteDeathNotes,
    whitePos: stateData.whitePosNotes,
    twoDeath: stateData.twoDeathNotes,
    twoPos: stateData.twoPosNotes,
    aianDeath: stateData.aianDeathNotes,
    aianPos: stateData.aianPosNotes,
    nhpiDeath: stateData.nhpiDeathNotes,
    nhpiPos: stateData.nhpiPosNotes,
    asianDeath: stateData.asianDeathNotes,
    asianPos: stateData.asianPosNotes,
    latinXDeath: stateData.latinXDeathNotes,
    latinXPos: stateData.latinXPosNotes,
    blackDeath: stateData.blackDeathNotes,
    blackPos: stateData.blackPosNotes,
  }
  notes = anhpiNotes(stateData, notes)
  notes = cautionNotes(stateData, notes)
  const groupedNotes = [...new Set(Object.values(notes))]
    .reverse()
    .filter(value => value && value.trim().length && value)

  const disparityExists =
    Object.keys(stateData).filter(
      field => field.search('DispFlag') > -1 && stateData[field],
    ).length > 0

  return (
    <div>
      <div className={stateStyle.stateOverview}>
        <PercentageOverview
          stateName={state.name}
          dataType="race and ethnicity"
          hospitalizationPercent={state.knownRaceEthHosp}
          testPercent={state.knownRaceEthTest}
          casePercent={state.knownRaceEthPos}
          deathPercent={state.knownRaceEthDeath}
          className={stateStyle.totals}
        />
        <div className={stateStyle.note}>
          <HeaderSorter stateName={state.name} stateReports="race/ethnicity" />
        </div>
      </div>
      <h3 className={stateStyle.tableTitle}>
        Cases and deaths by race/ethnicity
      </h3>
      <RaceTable
        data={stateData}
        type="Race/ethnicity"
        notes={notes}
        groupedNotes={groupedNotes}
        noPositives={!stateData.anyPosData}
        noDeaths={!stateData.anyDeathData}
        isCombined
      />
      <TableNotes
        state={stateData.state}
        stateName={stateData.name}
        groupedNotes={groupedNotes}
        disparityExists={disparityExists}
        hispanicLatinxNote
      />
    </div>
  )
}

export default StateCombined
