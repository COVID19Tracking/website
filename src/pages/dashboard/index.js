import React from 'react'
import Layout from '../../components/layout'
import Map from './_Map'

const DashboardPage = () => (
  <Layout title="Visual Dashboard">
    <p>Test</p>
    <Map />
  </Layout>
)

export default DashboardPage
