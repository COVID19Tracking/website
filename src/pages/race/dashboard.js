import React from 'react'
import { graphql } from 'gatsby'
import Layout from '~components/layout'

import StateNav from '~components/pages/race/dashboard/state-nav'
import LongContent from '~components/common/long-content'

import States from '~components/pages/race/dashboard/states'
import UsOverview from '~components/pages/race/dashboard/us-overview'
import Preamble from '~components/pages/race/dashboard/preamble'

const RaceDashboardPage = ({ data }) => {
  const stateList = []
  data.allCovidRaceDataSeparate.nodes.forEach(state => {
    stateList.push(state)
  })
  data.allCovidRaceDataCombined.nodes.forEach(state => {
    stateList.push(state)
  })
  return (
    <Layout
      title="Racial Data Dashboard"
      returnLinks={[{ link: '/race', title: 'Racial Data Tracker' }]}
      path="/race/dashboard"
      socialCard={data.contentfulSocialCard}
    >
      <LongContent>
        <Preamble
          raceHeroSnippetHtml={
            data.raceHeroSnippet.childContentfulSnippetContentTextNode
              .childMarkdownRemark.html
          }
        />
        <UsOverview
          statesCasesCount={data.covidRaceDataHomepage.statesReportingCases}
          statesDeathsCount={data.covidRaceDataHomepage.statesReportingDeaths}
        />
      </LongContent>
      <StateNav
        title="Race and ethnicity data by state"
        stateList={stateList.sort((a, b) => (a.name < b.name ? -1 : 1))}
      />
      <States />
    </Layout>
  )
}

export default RaceDashboardPage

export const query = graphql`
  query {
    raceHeroSnippet: contentfulSnippet(slug: { eq: "race-hero-lede" }) {
      childContentfulSnippetContentTextNode {
        childMarkdownRemark {
          html
        }
      }
    }
    contentfulSocialCard(slug: { eq: "racial-data-tracker" }) {
      description {
        description
      }
      image {
        resize(width: 1200) {
          src
        }
      }
    }
    allCovidRaceDataSeparate {
      nodes {
        name
        state
      }
    }
    allCovidRaceDataCombined {
      nodes {
        name
        state
      }
    }
    covidRaceDataHomepage {
      statesReportingCases
      statesReportingDeaths
    }
    allCovidStateInfo {
      nodes {
        state
        name
      }
    }
  }
`
