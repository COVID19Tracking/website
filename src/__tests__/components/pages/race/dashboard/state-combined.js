import React from 'react'
import renderer from 'react-test-renderer'
import { useStaticQuery } from 'gatsby'
import { StateCombined } from '~components/pages/race/dashboard/state-combined'

beforeEach(() => {
  useStaticQuery.mockImplementation(() => ({
    disparityNote: {
      contentful_id: 'aNeWRaNdoMSTrinG',
      childContentfulSnippetContentTextNode: {
        childMarkdownRemark: {
          html: '<h2>Content</h2><p>More disparity snippet content</p>',
        },
      },
    },
    comparibleNote: {
      contentful_id: 'RaNdoMSTrinG',
      childContentfulSnippetContentTextNode: {
        childMarkdownRemark: {
          html: '<h2>Content</h2><p>More comparible snippet content</p>',
        },
      },
    },
  }))
})

describe('Components : Pages : Race : Dashboard : State Combined', () => {
  it('renders correctly', () => {
    const state = {
      aianDeathCaution: true,
      aianDeathDispFlag: false,
      aianDeathNotes: 'Sample Sample aianDeath notes.',
      aianPctDeath: 'aianPctDeath',
      aianPctPop: 0.786,
      aianPctPos: 0.599,
      aianPosCaution: true,
      aianPosDispFlag: false,
      aianPosNotes: 'Sample Sample aianPos notes.',
      anyDeathData: true,
      anyPosData: true,
      asianDeathCaution: true,
      asianDeathDispFlag: false,
      asianDeathNotes: null,
      asianPctDeath: 'asianPctDeath',
      asianPctPop: 0.849,
      asianPctPos: 0.524,
      asianPosCaution: true,
      asianPosDispFlag: false,
      asianPosNotes: 'Sample Sample asianPos notes.',
      blackDeathCaution: true,
      blackDeathDispFlag: false,
      blackDeathNotes: null,
      blackPctDeath: 'blackPctDeath',
      blackPctPop: 0.22,
      blackPctPos: 0.025,
      blackPosCaution: true,
      blackPosDispFlag: false,
      blackPosNotes: 'Sample Sample blackPos notes.',
      latinXDeathDispFlag: false,
      latinXDeathNotes: 'Sample Sample latinXDeath notes.',
      latinXPctDeath: 'latinXPctDeath',
      latinXPctPop: 0.745,
      latinXPctPos: 0.121,
      latinXPosCaution: true,
      latinXPosDispFlag: false,
      latinXPosNotes: 'Sample Sample latinXPos notes.',
      name: 'Alabama',
      nhpiDeathCaution: true,
      nhpiDeathDispFlag: false,
      nhpiDeathNotes: 'Sample Sample nhpiDeath notes.',
      nhpiPctDeath: 'nhpiPctDeath',
      nhpiPctPop: 0.36,
      nhpiPctPos: 0.263,
      nhpiPosCaution: true,
      nhpiPosDispFlag: false,
      nhpiPosNotes: 'Sample Sample nhpiPos notes.',
      otherDeathCaution: true,
      otherDeathDispFlag: false,
      otherDeathNotes: 'Sample Sample otherDeath notes.',
      otherPctDeath: 0.123,
      otherPctPop: 0.31,
      otherPctPos: 0.857,
      otherPosCaution: true,
      otherPosDispFlag: false,
      otherPosNotes: 'Sample Sample otherPos notes.',
      state: 'AL',
      twoDeathCaution: true,
      twoDeathDispFlag: false,
      twoDeathNotes: 'Sample Sample twoDeath notes.',
      twoPctDeath: 'twoPctDeath',
      twoPctPop: 0.989,
      twoPctPos: 0.379,
      twoPosCaution: true,
      twoPosDispFlag: false,
      twoPosNotes: 'Sample Sample twoPos notes.',
      whiteDeathCaution: true,
      whiteDeathDispFlag: false,
      whiteDeathNotes: null,
      whitePctDeath: 'whitePctDeath',
      whitePctPop: 0.033,
      whitePctPos: 0.787,
      whitePosCaution: true,
      whitePosDispFlag: false,
      whitePosNotes: 'Sample Sample whitePos notes.',
    }
    const tree = renderer.create(<StateCombined state={state} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
