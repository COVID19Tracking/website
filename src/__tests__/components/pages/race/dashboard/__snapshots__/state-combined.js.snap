// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components : Pages : Race : Dashboard : State Combined renders correctly 1`] = `
<div>
  <div
    className="stateOverview"
  >
    <div
      className="note"
    >
      <div
        className="container"
      >
        <p>
          The following tables reflect only those cases and deaths where
           
          race/ethnicity
           is known and reported by 
          Alabama
          . If this stateâ€™s reporting percentages are low, interpret with caution.
        </p>
      </div>
    </div>
  </div>
  <h3
    className="tableTitle"
  >
    Cases and deaths by race/ethnicity
  </h3>
  <div>
    <table
      className="table"
    >
      <thead>
        <tr>
          <th
            className="group"
            scope="col"
          >
            <span
              role="text"
            >
              Race/ethnicity
            </span>
          </th>
          <th
            className="borderLeft percent"
            scope="col"
          >
            <span
              role="text"
            >
              Percentage of population
            </span>
          </th>
          <th
            className="percent"
            scope="col"
          >
            <span
              role="text"
            >
              Percentage of cases
              <div
                className="symbolSpacer"
              />
            </span>
          </th>
          <th
            className="percent"
            scope="col"
          >
            <span
              role="text"
            >
              Percentage of deaths
              <div
                className="symbolSpacer"
              />
            </span>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <th
            className="group"
            scope="row"
          >
            <span
              role="text"
            >
              Black or African American alone
            </span>
          </th>
          <td
            className="borderLeft"
          >
            22
            %
          </td>
          <td
            className=""
          >
            <span
              data-reach-tooltip-trigger=""
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseMove={[Function]}
            >
              <span
                className="highlight"
              >
                3
                %
              </span>
            </span>
            <div
              className="symbolSpacer"
            >
              <span
                className="note"
                data-reach-tooltip-trigger=""
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseMove={[Function]}
              >
                <a
                  href="#notes-al"
                >
                  <span
                    className="a11y-only"
                  >
                    This data should not be compared with percentage of the the population. Sample Sample blackPos notes.
                  </span>
                  <span>
                    1
                  </span>
                </a>
              </span>
            </div>
          </td>
          <td
            className=""
          >
            <span
              data-reach-tooltip-trigger=""
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseMove={[Function]}
            >
              <span
                className="highlight"
              >
                0
                %
              </span>
            </span>
            <div
              className="symbolSpacer"
            >
              <span
                className="note"
                data-reach-tooltip-trigger=""
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseMove={[Function]}
              >
                <a
                  href="#notes-al"
                >
                  <span
                    className="a11y-only"
                  >
                    This data should not be compared with percentage of the the population. 
                  </span>
                  <span>
                    12
                  </span>
                </a>
              </span>
            </div>
          </td>
        </tr>
        <tr>
          <th
            className="group"
            scope="row"
          >
            <span
              role="text"
            >
              Hispanic or Latino *
            </span>
          </th>
          <td
            className="borderLeft"
          >
            75
            %
          </td>
          <td
            className=""
          >
            <span
              data-reach-tooltip-trigger=""
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseMove={[Function]}
            >
              <span
                className="highlight"
              >
                12
                %
              </span>
            </span>
            <div
              className="symbolSpacer"
            >
              <span
                className="note"
                data-reach-tooltip-trigger=""
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseMove={[Function]}
              >
                <a
                  href="#notes-al"
                >
                  <span
                    className="a11y-only"
                  >
                    This data should not be compared with percentage of the the population. Sample Sample latinXPos notes.
                  </span>
                  <span>
                    2
                  </span>
                </a>
              </span>
            </div>
          </td>
          <td
            className=""
          >
            <span
              data-reach-tooltip-trigger=""
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseMove={[Function]}
            >
              <span
                className="highlight"
              >
                0
                %
              </span>
            </span>
            <div
              className="symbolSpacer"
            >
              <span
                className="note"
                data-reach-tooltip-trigger=""
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseMove={[Function]}
              >
                <a
                  href="#notes-al"
                >
                  <span
                    className="a11y-only"
                  >
                    Sample Sample latinXDeath notes.
                  </span>
                  <span>
                    3
                  </span>
                </a>
              </span>
            </div>
          </td>
        </tr>
        <tr>
          <th
            className="group"
            scope="row"
          >
            <span
              role="text"
            >
              Asian alone
            </span>
          </th>
          <td
            className="borderLeft"
          >
            85
            %
          </td>
          <td
            className=""
          >
            <span
              data-reach-tooltip-trigger=""
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseMove={[Function]}
            >
              <span
                className="highlight"
              >
                52
                %
              </span>
            </span>
            <div
              className="symbolSpacer"
            >
              <span
                className="note"
                data-reach-tooltip-trigger=""
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseMove={[Function]}
              >
                <a
                  href="#notes-al"
                >
                  <span
                    className="a11y-only"
                  >
                    This data should not be compared with percentage of the the population. Sample Sample asianPos notes.
                  </span>
                  <span>
                    4
                  </span>
                </a>
              </span>
            </div>
          </td>
          <td
            className=""
          >
            <span
              data-reach-tooltip-trigger=""
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseMove={[Function]}
            >
              <span
                className="highlight"
              >
                0
                %
              </span>
            </span>
            <div
              className="symbolSpacer"
            >
              <span
                className="note"
                data-reach-tooltip-trigger=""
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseMove={[Function]}
              >
                <a
                  href="#notes-al"
                >
                  <span
                    className="a11y-only"
                  >
                    This data should not be compared with percentage of the the population. 
                  </span>
                  <span>
                    12
                  </span>
                </a>
              </span>
            </div>
          </td>
        </tr>
        <tr>
          <th
            className="group"
            scope="row"
          >
            <span
              role="text"
            >
              Native Hawaiian and Pacific Islander alone
            </span>
          </th>
          <td
            className="borderLeft"
          >
            36
            %
          </td>
          <td
            className=""
          >
            <span
              data-reach-tooltip-trigger=""
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseMove={[Function]}
            >
              <span
                className="highlight"
              >
                26
                %
              </span>
            </span>
            <div
              className="symbolSpacer"
            >
              <span
                className="note"
                data-reach-tooltip-trigger=""
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseMove={[Function]}
              >
                <a
                  href="#notes-al"
                >
                  <span
                    className="a11y-only"
                  >
                    This data should not be compared with percentage of the the population. Sample Sample nhpiPos notes.
                  </span>
                  <span>
                    5
                  </span>
                </a>
              </span>
            </div>
          </td>
          <td
            className=""
          >
            <span
              data-reach-tooltip-trigger=""
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseMove={[Function]}
            >
              <span
                className="highlight"
              >
                0
                %
              </span>
            </span>
            <div
              className="symbolSpacer"
            >
              <span
                className="note"
                data-reach-tooltip-trigger=""
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseMove={[Function]}
              >
                <a
                  href="#notes-al"
                >
                  <span
                    className="a11y-only"
                  >
                    This data should not be compared with percentage of the the population. Sample Sample nhpiDeath notes.
                  </span>
                  <span>
                    6
                  </span>
                </a>
              </span>
            </div>
          </td>
        </tr>
        <tr>
          <th
            className="group"
            scope="row"
          >
            <span
              role="text"
            >
              American Indian or Alaska Native alone
            </span>
          </th>
          <td
            className="borderLeft"
          >
            79
            %
          </td>
          <td
            className=""
          >
            <span
              data-reach-tooltip-trigger=""
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseMove={[Function]}
            >
              <span
                className="highlight"
              >
                60
                %
              </span>
            </span>
            <div
              className="symbolSpacer"
            >
              <span
                className="note"
                data-reach-tooltip-trigger=""
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseMove={[Function]}
              >
                <a
                  href="#notes-al"
                >
                  <span
                    className="a11y-only"
                  >
                    This data should not be compared with percentage of the the population. Sample Sample aianPos notes.
                  </span>
                  <span>
                    7
                  </span>
                </a>
              </span>
            </div>
          </td>
          <td
            className=""
          >
            <span
              data-reach-tooltip-trigger=""
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseMove={[Function]}
            >
              <span
                className="highlight"
              >
                0
                %
              </span>
            </span>
            <div
              className="symbolSpacer"
            >
              <span
                className="note"
                data-reach-tooltip-trigger=""
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseMove={[Function]}
              >
                <a
                  href="#notes-al"
                >
                  <span
                    className="a11y-only"
                  >
                    This data should not be compared with percentage of the the population. Sample Sample aianDeath notes.
                  </span>
                  <span>
                    8
                  </span>
                </a>
              </span>
            </div>
          </td>
        </tr>
        <tr>
          <th
            className="group"
            scope="row"
          >
            <span
              role="text"
            >
              Two or more races
            </span>
          </th>
          <td
            className="borderLeft"
          >
            99
            %
          </td>
          <td
            className=""
          >
            <span
              data-reach-tooltip-trigger=""
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseMove={[Function]}
            >
              <span
                className="highlight"
              >
                38
                %
              </span>
            </span>
            <div
              className="symbolSpacer"
            >
              <span
                className="note"
                data-reach-tooltip-trigger=""
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseMove={[Function]}
              >
                <a
                  href="#notes-al"
                >
                  <span
                    className="a11y-only"
                  >
                    This data should not be compared with percentage of the the population. Sample Sample twoPos notes.
                  </span>
                  <span>
                    9
                  </span>
                </a>
              </span>
            </div>
          </td>
          <td
            className=""
          >
            <span
              data-reach-tooltip-trigger=""
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseMove={[Function]}
            >
              <span
                className="highlight"
              >
                0
                %
              </span>
            </span>
            <div
              className="symbolSpacer"
            >
              <span
                className="note"
                data-reach-tooltip-trigger=""
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseMove={[Function]}
              >
                <a
                  href="#notes-al"
                >
                  <span
                    className="a11y-only"
                  >
                    This data should not be compared with percentage of the the population. Sample Sample twoDeath notes.
                  </span>
                  <span>
                    10
                  </span>
                </a>
              </span>
            </div>
          </td>
        </tr>
        <tr>
          <th
            className="group"
            scope="row"
          >
            <span
              role="text"
            >
              White alone
            </span>
          </th>
          <td
            className="borderLeft"
          >
            3
            %
          </td>
          <td
            className=""
          >
            <span
              data-reach-tooltip-trigger=""
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseMove={[Function]}
            >
              <span
                className="highlight"
              >
                79
                %
              </span>
            </span>
            <div
              className="symbolSpacer"
            >
              <span
                className="note"
                data-reach-tooltip-trigger=""
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseMove={[Function]}
              >
                <a
                  href="#notes-al"
                >
                  <span
                    className="a11y-only"
                  >
                    This data should not be compared with percentage of the the population. Sample Sample whitePos notes.
                  </span>
                  <span>
                    11
                  </span>
                </a>
              </span>
            </div>
          </td>
          <td
            className=""
          >
            <span
              data-reach-tooltip-trigger=""
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseMove={[Function]}
            >
              <span
                className="highlight"
              >
                0
                %
              </span>
            </span>
            <div
              className="symbolSpacer"
            >
              <span
                className="note"
                data-reach-tooltip-trigger=""
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseMove={[Function]}
              >
                <a
                  href="#notes-al"
                >
                  <span
                    className="a11y-only"
                  >
                    This data should not be compared with percentage of the the population. 
                  </span>
                  <span>
                    12
                  </span>
                </a>
              </span>
            </div>
          </td>
        </tr>
        <tr>
          <th
            className="group"
            scope="row"
          >
            <span
              role="text"
            >
              Some other race alone
            </span>
          </th>
          <td
            className="borderLeft"
          >
            31
            %
          </td>
          <td
            className=""
          >
            <span
              data-reach-tooltip-trigger=""
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseMove={[Function]}
            >
              <span
                className="highlight"
              >
                86
                %
              </span>
            </span>
            <div
              className="symbolSpacer"
            >
              <span
                className="note"
                data-reach-tooltip-trigger=""
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseMove={[Function]}
              >
                <a
                  href="#notes-al"
                >
                  <span
                    className="a11y-only"
                  >
                    This data should not be compared with percentage of the the population. Sample Sample otherPos notes.
                  </span>
                  <span>
                    13
                  </span>
                </a>
              </span>
            </div>
          </td>
          <td
            className=""
          >
            <span
              data-reach-tooltip-trigger=""
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseMove={[Function]}
            >
              <span
                className="highlight"
              >
                12
                %
              </span>
            </span>
            <div
              className="symbolSpacer"
            >
              <span
                className="note"
                data-reach-tooltip-trigger=""
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseMove={[Function]}
              >
                <a
                  href="#notes-al"
                >
                  <span
                    className="a11y-only"
                  >
                    This data should not be compared with percentage of the the population. Sample Sample otherDeath notes.
                  </span>
                  <span>
                    14
                  </span>
                </a>
              </span>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <h4
    className="header"
    id="notes-al"
  >
    Notes
     
    <span
      className="a11y-only"
    >
       for 
      Alabama
    </span>
  </h4>
  <ul
    className="list"
  >
    <li>
      <span
        className="emptyNote"
      >
        *
      </span>
       Hispanic or Latino ethnicity, any race. All other race categories in this table are defined as Not Hispanic or Latino.
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          1
        </span>
      </span>
       
      This data should not be compared with percentage of the the population. Sample Sample blackPos notes.
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          2
        </span>
      </span>
       
      This data should not be compared with percentage of the the population. Sample Sample latinXPos notes.
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          3
        </span>
      </span>
       
      Sample Sample latinXDeath notes.
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          4
        </span>
      </span>
       
      This data should not be compared with percentage of the the population. Sample Sample asianPos notes.
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          5
        </span>
      </span>
       
      This data should not be compared with percentage of the the population. Sample Sample nhpiPos notes.
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          6
        </span>
      </span>
       
      This data should not be compared with percentage of the the population. Sample Sample nhpiDeath notes.
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          7
        </span>
      </span>
       
      This data should not be compared with percentage of the the population. Sample Sample aianPos notes.
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          8
        </span>
      </span>
       
      This data should not be compared with percentage of the the population. Sample Sample aianDeath notes.
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          9
        </span>
      </span>
       
      This data should not be compared with percentage of the the population. Sample Sample twoPos notes.
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          10
        </span>
      </span>
       
      This data should not be compared with percentage of the the population. Sample Sample twoDeath notes.
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          11
        </span>
      </span>
       
      This data should not be compared with percentage of the the population. Sample Sample whitePos notes.
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          12
        </span>
      </span>
       
      This data should not be compared with percentage of the the population. 
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          13
        </span>
      </span>
       
      This data should not be compared with percentage of the the population. Sample Sample otherPos notes.
    </li>
    <li>
      <span
        aria-hidden={true}
        className="note unlinked"
      >
        <span>
          14
        </span>
      </span>
       
      This data should not be compared with percentage of the the population. Sample Sample otherDeath notes.
    </li>
  </ul>
</div>
`;
